splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceSystemProcedures.java

```
@@ -1129,9 +1136,45 @@ protected void augmentProcedureMap(DataDictionary dict, Map procedures) throws S
                             .isDeterministic(false).ownerClass(GenericSpliceFunctions.class.getCanonicalName())
                             .build()
             ));
+
+            // Add public static methods from SysfunProviders that are on the system classpath.
+            for (SysfunProvider provider : ServiceLoader.load(SysfunProvider.class)) {
+                for (Method m : provider.getMethods()) {
+                    if (!Modifier.isStatic(m.getModifiers()) || !Modifier.isPublic(m.getModifiers()))
+                        continue;
```

This should make a horrible noise - otherwise we run the risk of spending a long time debugging a silent failure to register a function.